hw8_sysos/ls.py

```diff
@@ -0,0 +1,46 @@
+#! /usr/bin/env python3
+# ls_function_analogue
+import os
+import sys
+import argparse
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        usage='./ls.py  [OPTION]... [FILE]...',
+        description='''List information about the FILEs (the current directory by default).
+        Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.''')
+
+    parser.add_argument('-a', '--all', action='store_true',
+                        help='do not ignore entries starting with .')
+    parser.add_argument('directory_path', type=str, default='.', nargs='*')
+
+    args = parser.parse_args()
+
+    target_dir = args.directory_path
+
+
+    # multiple directrories may be indicated, iterate over them
+    for directory in args.directory_path:
+        # check if a directory exists
+        if not os.path.exists(directory):
+            sys.stderr.write(f'ls: cannot access \'{directory}\': No such file or directory\n')
+            raise sys.exit(0)

```

Comment:
1. Почему raise?
2. Если это ошибка, то надо `sys.exit(1)`
3. До raise дело вообще не дойдёт, программа закончится на sys.exit
```suggestion
            sys.exit(1)

```